src/pip/_internal/commands/wheel.py

```diff
@@ -177,6 +186,19 @@ def run(self, options, args):
                         raise CommandError(
                             "Failed to build one or more wheels"
                         )
+                    if wb.path_to_wheelnames is not None:
+                        entries_to_save = wb.wheel_filenames[::]
+                        for req in requirement_set.requirements.values():
+                            if req.link.filename.endswith('whl'):
+                                entries_to_save.append(req.link.filename)
+
+                        with open(wb.path_to_wheelnames, 'w') as file:
```

Comment:
what happens if the file is not writable?

Reply:
Program would fail. So i should instead printed that filenames would not be saved. Then should I ask if the user want to give another path ?

Reply:
Does the program give an ugly traceback or some rather nice error message?